tests/python/unittest/test_numpy_op.py

```diff
@@ -1519,6 +1519,58 @@ def hybrid_forward(self, F, a):
                         assert_almost_equal(mx_out.asnumpy(), np_out, rtol=1e-3, atol=1e-5)
 
 
+@with_seed()
+@use_np
+def test_np_diff():
+    def np_diff_backward(ograd, n, axis):
+        res = ograd
+        for i in range(n):
+            res = _np.negative(_np.diff(res, n=1, axis=axis, prepend=0, append=0))
+        return res
+
+    class TestDiff(HybridBlock):
+        def __init__(self, n=1, axis=-1):
+            super(TestDiff, self).__init__()
+            self._n = n
+            self._axis = axis
+
+        def hybrid_forward(self, F, a):
+            return F.np.diff(a, n=self._n, axis=self._axis)
+
+    shapes = [tuple(random.randrange(10) for i in range(random.randrange(6))) for j in range(5)]
```

Comment:
Please design static shapes for testing so that we could guarantee all corner cases are covered.

Reply:
Hi, the random shape generator should have covered the corner cases, including zero dim, 1d to 5d input shape. Actually, the codes have been tested by static shapes and the random generator here is for shorter testing time.
Do you think it's necessary to change back still? If yes, I'll do it.

Reply:
I prefer to have a static shape list tho. Here are my concerns, and you are welcome to let me know if those look not an issue in your opinion:

- This code may be maintained or improved by others in the future, but anyone new to this code will get confused about what tests it has covered.
- If the code is modified in the future that fails to cover certain cases, it is hard to capture because re-run the test may not give you the same shape. We should keep the unit test result deterministic.

Reply:
I agreed your opinions and that's what I did before, but the present random cases generator should be concise and easy to understand. 
The tests are switched to random cases mainly for the balance of time cost and test coverage, which is advised by @haojin2 , could you reach an agreement?

Reply:
I don't quite get the point about the balance of time cost and test coverage tho. It seems to me that declaring a static list (i.e., shape = [[...], [...], ...]) won't cost more than calling random generator on the fly. I would even prefer the way that you generate a list of shapes randomly and hardcoded the generated list in this unit test. On the other hand, I'm also fine if you and @haojin2 also agreed with the random generator way in case you guys have other concerns I missed.